# **Earthquake Analysis**
## *ETL Notebook* 


## Objectives

* Write your notebook objective here, for example, "Fetch data from Kaggle and save as raw data", or "engineer features for modelling"

* Import the intial .csv file to a pandas dataframe for inspection.

* Diagnose and make any necessary Transformations

* Engineer useful features for classificaiton and Machine Learning.

* Cement the made changes into a new dataframe, to be exported to a new .csv 

## Inputs

* The inital 'earthquake_1995-2023.csv'

* Libraries containing the tools needed. Pandas, numpy, matplotlib and seaborn.


## Outputs

* REMEMBER TO FILL THIS IN






---

# Change working directory

* We are assuming you will store the notebooks in a subfolder, therefore when running the notebook in the editor, you will need to change the working directory

We need to change the working directory from its current folder to its parent folder
* We access the current directory with os.getcwd()

In [1]:
import os
current_dir = os.getcwd()
current_dir

'c:\\Users\\NateM\\Desktop\\CI_DA\\Earthquake_Analysis\\jupyter_notebooks'

We want to make the parent of the current directory the new current directory
* os.path.dirname() gets the parent directory
* os.chir() defines the new current directory

In [2]:
os.chdir(os.path.dirname(current_dir))
print("You set a new current directory")

You set a new current directory


Confirm the new current directory

In [3]:
current_dir = os.getcwd()
current_dir

'c:\\Users\\NateM\\Desktop\\CI_DA\\Earthquake_Analysis'

# Import block

These are the libraries required to process and inspect the data. 

Additionally I have an acompanying python file, HelperFuncs. 
Which contains refactored processess as callable functions.

In [4]:
# Librairies needed to perform analysis

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns


# Acompanying python file  
from HelperFuncs import InitialInspect

# Section  1 -  Extraction and initial inspections

Let's load the data and see what we've got.

In [5]:
# Storing the csv into a workable variable 
df_raw = pd.read_csv("Dataset/Raw/earthquake_1995-2023.csv")

# Calling a helper function 
# Output will be truncated, view as a scrollable element
InitialInspect(df_raw)

Initial shape: (Row x Col) (1000, 19)
Columns data types:
 title         object
magnitude    float64
date_time     object
cdi            int64
mmi            int64
alert         object
tsunami        int64
sig            int64
net           object
nst            int64
dmin         float64
gap          float64
magType       object
depth        float64
latitude     float64
longitude    float64
location      object
continent     object
country       object
dtype: object
Missing value counts: 
title: 0.
magnitude: 0.
date_time: 0.
cdi: 0.
mmi: 0.
alert: 551.
tsunami: 0.
sig: 0.
net: 0.
nst: 0.
dmin: 0.
gap: 0.
magType: 0.
depth: 0.
latitude: 0.
longitude: 0.
location: 6.
continent: 716.
country: 349.
Duplicate counts: 
title: 16 duplicates.
magnitude: 975 duplicates.
date_time: 10 duplicates.
cdi: 990 duplicates.
mmi: 990 duplicates.
alert: 995 duplicates.
tsunami: 998 duplicates.
sig: 644 duplicates.
net: 989 duplicates.
nst: 655 duplicates.
dmin: 598 duplicates.
gap: 741 duplicates.
magT

,magnitude,cdi,mmi,tsunami,sig,nst,dmin,gap,depth,latitude,longitude
count,1000.000000,1000.000000,1000.00000,1000.000000,1000.000000,1000.000000,1000.000000,1000.000000,1000.000000,1000.000000,1000.000000
mean,6.940150,3.605000,6.02700,0.325000,847.915000,193.918000,1.125174,20.926290,74.612541,4.315554,51.486576
std,0.438148,3.328972,1.43399,0.468609,301.802632,239.045858,2.073164,24.415895,130.812590,26.633320,117.478302
min,6.500000,0.000000,1.00000,0.000000,650.000000,0.000000,0.000000,0.000000,2.700000,-61.848400,-179.968000
25%,6.600000,0.000000,5.00000,0.000000,691.000000,0.000000,0.000000,0.000000,16.000000,-13.518500,-71.694450
50%,6.800000,4.000000,6.00000,0.000000,744.000000,0.000000,0.000000,18.000000,29.000000,-2.443500,107.791000
75%,7.100000,7.000000,7.00000,1.000000,874.250000,403.000000,1.549250,27.000000,55.000000,25.167250,148.364750
max,9.100000,9.000000,10.00000,1.000000,2910.000000,934.000000,17.654000,239.000000,670.810000,71.631200,179.662000


First 10 rows


,title,magnitude,date_time,cdi,mmi,alert,tsunami,sig,net,nst,dmin,gap,magType,depth,latitude,longitude,location,continent,country
0,"M 6.5 - 42 km W of Sola, Vanuatu",6.5,16-08-2023 12:47,7,4,green,0,657,us,114,7.177000,25.0,mww,192.955,-13.8814,167.1580,"Sola, Vanuatu",NaN,Vanuatu
1,"M 6.5 - 43 km S of Intipucá, El Salvador",6.5,19-07-2023 00:22,8,6,yellow,0,775,us,92,0.679000,40.0,mww,69.727,12.8140,-88.1265,"Intipucá, El Salvador",NaN,NaN
2,"M 6.6 - 25 km ESE of Loncopué, Argentina",6.6,17-07-2023 03:05,7,5,green,0,899,us,70,1.634000,28.0,mww,171.371,-38.1911,-70.3731,"Loncopué, Argentina",South America,Argentina
3,"M 7.2 - 98 km S of Sand Point, Alaska",7.2,16-07-2023 06:48,6,6,green,1,860,us,173,0.907000,36.0,mww,32.571,54.3844,-160.6990,"Sand Point, Alaska",NaN,NaN
4,M 7.3 - Alaska Peninsula,7.3,16-07-2023 06:48,0,5,NaN,1,820,at,79,0.879451,172.8,Mi,21.000,54.4900,-160.7960,Alaska Peninsula,NaN,NaN
5,"M 6.6 - 277 km NNE of Codrington, Antigua and ...",6.6,10-07-2023 20:28,5,4,green,1,802,us,95,2.454000,37.0,mww,10.000,20.0196,-61.0955,"Codrington, Antigua and Barbuda",NaN,NaN
6,M 6.9 - Tonga,6.9,02-07-2023 10:27,4,4,green,1,741,us,136,1.179000,23.0,mww,229.000,-17.8530,-174.9370,Tonga,NaN,NaN
7,M 7.2 - south of the Fiji Islands,7.2,15-06-2023 18:06,8,6,green,1,804,us,85,2.590000,24.0,mww,167.404,-22.9824,-177.2080,the Fiji Islands,NaN,NaN
8,M 6.6 - Panama-Colombia border region,6.6,25-05-2023 03:05,6,6,green,1,733,us,50,2.163000,129.0,mww,10.000,8.8894,-77.1231,Panama-Colombia border region,NaN,Colombia
9,M 7.1 - southeast of the Loyalty Islands,7.1,20-05-2023 01:51,3,4,green,1,777,us,98,2.812000,56.0,mww,35.981,-23.0622,170.4560,the Loyalty Islands,NaN,NaN


This reports shows us that:

* The date column will need its type changing from object to be DateTime
* 551 missing values for "alert", 6 for "loaction", 716 for "continent" and 349 for "country"
* The nature of our duplicates are genuine repeat information

Next is to check the unqiue value counts for the columns with string values. This is to check for any non-uniform entries.

In [6]:
# Storing columns in as a single list 
to_value_check = ["title"], ["alert"], ["net"], ["magType"], ["location"], ["continent"], ["country"]

# Looping over the list, printing the results
# Output will be truncated, view as a scrollable element
for column in to_value_check:
    print("\nValue counts for: ")
    print(df_raw[column].value_counts(dropna=False))
    print(f"Total values: {df_raw[column].value_counts().sum()}")
    print("-" * 50)


Value counts for: 
title                                          
M 6.9 -                                            3
M 6.5 - Kermadec Islands, New Zealand              3
M 7.1 - South Sandwich Islands region              2
M 6.9 - South Sandwich Islands region              2
M 6.7 - Sea of Okhotsk                             2
                                                  ..
M 6.7 - 128 km SSE of Panguna, Papua New Guinea    1
M 6.7 - 13 km N of Do?anyol, Turkey                1
M 6.7 - 131 km SE of Nookat, Kyrgyzstan            1
M 6.7 - 132 km E of Tadine, New Caledonia          1
M 9.1 - 2011 Great Tohoku Earthquake, Japan        1
Name: count, Length: 984, dtype: int64
Total values: 1000
--------------------------------------------------

Value counts for: 
alert 
NaN       551
green     353
yellow     59
orange     24
red        13
Name: count, dtype: int64
Total values: 449
--------------------------------------------------

Value counts for: 
net     
us          960
ak 

No uniformity needs to be imposed.

'Null-island Artifacts', are where system errors and API failures issue 0.0, for the longitude and latitude. The rows that do have that result will need to investigated against other available location data and imputed.

In [15]:
print("Latitude '0.0' count:")
print((df_raw["latitude"]==0.0).sum())
print("Longiitude '0.0' count:")
print((df_raw["longitude"]==0.0).sum())

Latitude '0.0' count:
0
Longiitude '0.0' count:
0


We have no Null-island artifacts.

We can move on to handling the cleaning of the data.

---

# Section 2 - Transformations

The "date_time" columns' data type needs to be changed to DateTime.

In [7]:
# Before
print("Column type is: ")
print(df_raw["date_time"].dtypes)

# Conversion
df_raw["date_time"] = pd.to_datetime(df_raw["date_time"])

# After
print("Column type is: ")
print(df_raw["date_time"].dtypes)

df_raw.head(5)

Column type is: 
object
Column type is: 
datetime64[ns]


C:\Users\NateM\AppData\Local\Temp\ipykernel_22100\213738063.py:6: UserWarning: Parsing dates in %d-%m-%Y %H:%M format when dayfirst=False (the default) was specified. Pass `dayfirst=True` or specify a format to silence this warning.
  df_raw["date_time"] = pd.to_datetime(df_raw["date_time"])


,title,magnitude,date_time,cdi,mmi,alert,tsunami,sig,net,nst,dmin,gap,magType,depth,latitude,longitude,location,continent,country
0,"M 6.5 - 42 km W of Sola, Vanuatu",6.5,2023-08-16 12:47:00,7,4,green,0,657,us,114,7.177000,25.0,mww,192.955,-13.8814,167.1580,"Sola, Vanuatu",NaN,Vanuatu
1,"M 6.5 - 43 km S of Intipucá, El Salvador",6.5,2023-07-19 00:22:00,8,6,yellow,0,775,us,92,0.679000,40.0,mww,69.727,12.8140,-88.1265,"Intipucá, El Salvador",NaN,NaN
2,"M 6.6 - 25 km ESE of Loncopué, Argentina",6.6,2023-07-17 03:05:00,7,5,green,0,899,us,70,1.634000,28.0,mww,171.371,-38.1911,-70.3731,"Loncopué, Argentina",South America,Argentina
3,"M 7.2 - 98 km S of Sand Point, Alaska",7.2,2023-07-16 06:48:00,6,6,green,1,860,us,173,0.907000,36.0,mww,32.571,54.3844,-160.6990,"Sand Point, Alaska",NaN,NaN
4,M 7.3 - Alaska Peninsula,7.3,2023-07-16 06:48:00,0,5,NaN,1,820,at,79,0.879451,172.8,Mi,21.000,54.4900,-160.7960,Alaska Peninsula,NaN,NaN


Now the missing values.

USGS (Unites States Geological Survey) only assigns PAGER (Promt Assessment of Global Earthquakes for Response) colour values to events that directly affect population or the economy. So the missing values for "alert" can simply be filled with 'unassigned'. 

Events taking place off-shore or between territories do not get assigned a country value when recorded. These can simply be labeled as 'off-shore. Likewise, the "continent" column can be filled in as 'off-shore' or 'oceanic'. This column could be dropped entirely as we posses the latitude and longitude values of the recorded instances and they can be used for cross-referencing their geographical locaiton. Similarly, "location" can be filled with 'uknown' or be dropped.

I will choose to preserve and impute. 

In [8]:
# Output will be truncated, view as a scrollable element
# Filling in "alert"
df_raw["alert"] = df_raw["alert"].fillna("unassigned")
print(df_raw["alert"].value_counts())
print("=" * 50)

# Filling in "country"
df_raw["country"] = df_raw["country"].fillna("off_shore")
print(df_raw["country"].value_counts())
print("=" * 50)

# Filling in "continent"
df_raw["continent"] = df_raw["continent"].fillna("off_shore")
print(df_raw["continent"].value_counts())
print("=" * 50)

# Filling in "loaction"
df_raw["location"] = df_raw["location"].fillna("off_shore")
print(df_raw["location"].value_counts())
print("=" * 50)



alert
unassigned    551
green         353
yellow         59
orange         24
red            13
Name: count, dtype: int64
country
off_shore                                                     349
Indonesia                                                     140
Papua New Guinea                                               81
Chile                                                          42
Vanuatu                                                        36
Mexico                                                         31
Japan                                                          26
Peru                                                           26
Solomon Islands                                                25
United States of America                                       24
Philippines                                                    23
Russia                                                         19
People's Republic of China                                     15
Ecuador     

With those handled we can move on to feature engineering.

# Section 3 - Feature Engineering

We want to see how magnitude (the raw power output) behaves with its depth. Higher magnitude events, occuring at deeper depths will have different effects than ones at a shallower depth, as the energy dissipates over the time it travels. 

Multiplying magnitude by depth will give our model a combined metric to measure how much is any, depth affects the magnitude.





In [9]:
# Making the feature
df_raw["mag_depth_interaction"] = (
    df_raw["magnitude"] * df_raw["depth"]
    ).round(2)

# View the dataframe to confirm 
df_raw.head(5)

,title,magnitude,date_time,cdi,mmi,alert,tsunami,sig,net,nst,dmin,gap,magType,depth,latitude,longitude,location,continent,country,mag_depth_interaction
0,"M 6.5 - 42 km W of Sola, Vanuatu",6.5,2023-08-16 12:47:00,7,4,green,0,657,us,114,7.177000,25.0,mww,192.955,-13.8814,167.1580,"Sola, Vanuatu",off_shore,Vanuatu,1254.21
1,"M 6.5 - 43 km S of Intipucá, El Salvador",6.5,2023-07-19 00:22:00,8,6,yellow,0,775,us,92,0.679000,40.0,mww,69.727,12.8140,-88.1265,"Intipucá, El Salvador",off_shore,off_shore,453.23
2,"M 6.6 - 25 km ESE of Loncopué, Argentina",6.6,2023-07-17 03:05:00,7,5,green,0,899,us,70,1.634000,28.0,mww,171.371,-38.1911,-70.3731,"Loncopué, Argentina",South America,Argentina,1131.05
3,"M 7.2 - 98 km S of Sand Point, Alaska",7.2,2023-07-16 06:48:00,6,6,green,1,860,us,173,0.907000,36.0,mww,32.571,54.3844,-160.6990,"Sand Point, Alaska",off_shore,off_shore,234.51
4,M 7.3 - Alaska Peninsula,7.3,2023-07-16 06:48:00,0,5,unassigned,1,820,at,79,0.879451,172.8,Mi,21.000,54.4900,-160.7960,Alaska Peninsula,off_shore,off_shore,153.30


Tsunami's need an earthquake to happen close the sea floor. This depth can vary, grouping those into bins makes comparing the relevance of the depth easier.

Additionally, an earthquake can't trigger a tsunami if it happens inland. An 'is_off_shore' column will act a filter for potential tsunami generation. 

In [10]:
# "Depth" into bins
df_raw["depth_cat"] = pd.cut(
    df_raw["depth"],
    bins=[-np.inf, 70, 300, np.inf],
    labels=["shallow", "Intermediate", "Deep"],
)

# Off-shore marker
df_raw["is_off_shore"] = df_raw["country"].isin(["off_shore"])

# View to confirm the changes
df_raw.head(10)

,title,magnitude,date_time,cdi,mmi,alert,tsunami,sig,net,nst,...,magType,depth,latitude,longitude,location,continent,country,mag_depth_interaction,depth_cat,is_off_shore
0,"M 6.5 - 42 km W of Sola, Vanuatu",6.5,2023-08-16 12:47:00,7,4,green,0,657,us,114,...,mww,192.955,-13.8814,167.1580,"Sola, Vanuatu",off_shore,Vanuatu,1254.21,Intermediate,False
1,"M 6.5 - 43 km S of Intipucá, El Salvador",6.5,2023-07-19 00:22:00,8,6,yellow,0,775,us,92,...,mww,69.727,12.8140,-88.1265,"Intipucá, El Salvador",off_shore,off_shore,453.23,shallow,True
2,"M 6.6 - 25 km ESE of Loncopué, Argentina",6.6,2023-07-17 03:05:00,7,5,green,0,899,us,70,...,mww,171.371,-38.1911,-70.3731,"Loncopué, Argentina",South America,Argentina,1131.05,Intermediate,False
3,"M 7.2 - 98 km S of Sand Point, Alaska",7.2,2023-07-16 06:48:00,6,6,green,1,860,us,173,...,mww,32.571,54.3844,-160.6990,"Sand Point, Alaska",off_shore,off_shore,234.51,shallow,True
4,M 7.3 - Alaska Peninsula,7.3,2023-07-16 06:48:00,0,5,unassigned,1,820,at,79,...,Mi,21.000,54.4900,-160.7960,Alaska Peninsula,off_shore,off_shore,153.30,shallow,True
5,"M 6.6 - 277 km NNE of Codrington, Antigua and ...",6.6,2023-07-10 20:28:00,5,4,green,1,802,us,95,...,mww,10.000,20.0196,-61.0955,"Codrington, Antigua and Barbuda",off_shore,off_shore,66.00,shallow,True
6,M 6.9 - Tonga,6.9,2023-07-02 10:27:00,4,4,green,1,741,us,136,...,mww,229.000,-17.8530,-174.9370,Tonga,off_shore,off_shore,1580.10,Intermediate,True
7,M 7.2 - south of the Fiji Islands,7.2,2023-06-15 18:06:00,8,6,green,1,804,us,85,...,mww,167.404,-22.9824,-177.2080,the Fiji Islands,off_shore,off_shore,1205.31,Intermediate,True
8,M 6.6 - Panama-Colombia border region,6.6,2023-05-25 03:05:00,6,6,green,1,733,us,50,...,mww,10.000,8.8894,-77.1231,Panama-Colombia border region,off_shore,Colombia,66.00,shallow,False
9,M 7.1 - southeast of the Loyalty Islands,7.1,2023-05-20 01:51:00,3,4,green,1,777,us,98,...,mww,35.981,-23.0622,170.4560,the Loyalty Islands,off_shore,off_shore,255.47,shallow,True


Machine Learning is going to need a target label for what is a high severity event. 

We'll combine a few of the meausures that are used to classify this.
- An mmi of 7 or greater
- A sig score of 875 or greater
- An alert value of orange or red

MMI (Modified Mercalli Intensity) is used to classify the localised effect of an earthquake, rated from 1-10.

Sig score (Significance) is figure produce combining magnitude, PAGER alert level, maximum intensity, and total number of felt reports (cdi). 
We're using 875 as that's the datasets 75% percentile, creating a clear benchmark for what is high severity.

An orange alert value means 'estimated fatalities between 100 & 999 or economic losses between $100m and $1b (national scale impact) and a red alert value means 'estimated fatalties 1000+ or economic losses of over $1b (international response required).


In [16]:
# Implimenting the rule

high_severity_alert = ["orange", "red"]

df_raw["high_severity"] = np.where(
    (df_raw["mmi"] >= 7)
    | (df_raw["sig"] >= 875)
    | df_raw["alert"].isin(high_severity_alert),
     1,
     0,
)

# View changes
df_raw.head(10)



,title,magnitude,date_time,cdi,mmi,alert,tsunami,sig,net,nst,...,location,continent,country,mag_depth_interaction,depth_cat,is_off_shore,high_severity,year,month,hour
0,"M 6.5 - 42 km W of Sola, Vanuatu",6.5,2023-08-16 12:47:00,7,4,green,0,657,us,114,...,"Sola, Vanuatu",off_shore,Vanuatu,1254.21,Intermediate,False,0,2023,8,12
1,"M 6.5 - 43 km S of Intipucá, El Salvador",6.5,2023-07-19 00:22:00,8,6,yellow,0,775,us,92,...,"Intipucá, El Salvador",off_shore,off_shore,453.23,shallow,True,0,2023,7,0
2,"M 6.6 - 25 km ESE of Loncopué, Argentina",6.6,2023-07-17 03:05:00,7,5,green,0,899,us,70,...,"Loncopué, Argentina",South America,Argentina,1131.05,Intermediate,False,1,2023,7,3
3,"M 7.2 - 98 km S of Sand Point, Alaska",7.2,2023-07-16 06:48:00,6,6,green,1,860,us,173,...,"Sand Point, Alaska",off_shore,off_shore,234.51,shallow,True,0,2023,7,6
4,M 7.3 - Alaska Peninsula,7.3,2023-07-16 06:48:00,0,5,unassigned,1,820,at,79,...,Alaska Peninsula,off_shore,off_shore,153.30,shallow,True,0,2023,7,6
5,"M 6.6 - 277 km NNE of Codrington, Antigua and ...",6.6,2023-07-10 20:28:00,5,4,green,1,802,us,95,...,"Codrington, Antigua and Barbuda",off_shore,off_shore,66.00,shallow,True,0,2023,7,20
6,M 6.9 - Tonga,6.9,2023-07-02 10:27:00,4,4,green,1,741,us,136,...,Tonga,off_shore,off_shore,1580.10,Intermediate,True,0,2023,7,10
7,M 7.2 - south of the Fiji Islands,7.2,2023-06-15 18:06:00,8,6,green,1,804,us,85,...,the Fiji Islands,off_shore,off_shore,1205.31,Intermediate,True,0,2023,6,18
8,M 6.6 - Panama-Colombia border region,6.6,2023-05-25 03:05:00,6,6,green,1,733,us,50,...,Panama-Colombia border region,off_shore,Colombia,66.00,shallow,False,0,2023,5,3
9,M 7.1 - southeast of the Loyalty Islands,7.1,2023-05-20 01:51:00,3,4,green,1,777,us,98,...,the Loyalty Islands,off_shore,off_shore,255.47,shallow,True,0,2023,5,1


Breaking down the datetime information into seperate columns will make them useful in Machine Learning for predicting patterns over seasons. Combined with location data it could also be useful for detection efforts.  

In [12]:
# Extract datetime info into seperate columns
df_raw["year"] = df_raw["date_time"].dt.year
df_raw["month"] = df_raw["date_time"].dt.month
df_raw["hour"] = df_raw["date_time"].dt.hour

# View changes 
df_raw.head(10)

,title,magnitude,date_time,cdi,mmi,alert,tsunami,sig,net,nst,...,location,continent,country,mag_depth_interaction,depth_cat,is_off_shore,high_severity,year,month,hour
0,"M 6.5 - 42 km W of Sola, Vanuatu",6.5,2023-08-16 12:47:00,7,4,green,0,657,us,114,...,"Sola, Vanuatu",off_shore,Vanuatu,1254.21,Intermediate,False,0,2023,8,12
1,"M 6.5 - 43 km S of Intipucá, El Salvador",6.5,2023-07-19 00:22:00,8,6,yellow,0,775,us,92,...,"Intipucá, El Salvador",off_shore,off_shore,453.23,shallow,True,0,2023,7,0
2,"M 6.6 - 25 km ESE of Loncopué, Argentina",6.6,2023-07-17 03:05:00,7,5,green,0,899,us,70,...,"Loncopué, Argentina",South America,Argentina,1131.05,Intermediate,False,1,2023,7,3
3,"M 7.2 - 98 km S of Sand Point, Alaska",7.2,2023-07-16 06:48:00,6,6,green,1,860,us,173,...,"Sand Point, Alaska",off_shore,off_shore,234.51,shallow,True,0,2023,7,6
4,M 7.3 - Alaska Peninsula,7.3,2023-07-16 06:48:00,0,5,unassigned,1,820,at,79,...,Alaska Peninsula,off_shore,off_shore,153.30,shallow,True,0,2023,7,6
5,"M 6.6 - 277 km NNE of Codrington, Antigua and ...",6.6,2023-07-10 20:28:00,5,4,green,1,802,us,95,...,"Codrington, Antigua and Barbuda",off_shore,off_shore,66.00,shallow,True,0,2023,7,20
6,M 6.9 - Tonga,6.9,2023-07-02 10:27:00,4,4,green,1,741,us,136,...,Tonga,off_shore,off_shore,1580.10,Intermediate,True,0,2023,7,10
7,M 7.2 - south of the Fiji Islands,7.2,2023-06-15 18:06:00,8,6,green,1,804,us,85,...,the Fiji Islands,off_shore,off_shore,1205.31,Intermediate,True,0,2023,6,18
8,M 6.6 - Panama-Colombia border region,6.6,2023-05-25 03:05:00,6,6,green,1,733,us,50,...,Panama-Colombia border region,off_shore,Colombia,66.00,shallow,False,0,2023,5,3
9,M 7.1 - southeast of the Loyalty Islands,7.1,2023-05-20 01:51:00,3,4,green,1,777,us,98,...,the Loyalty Islands,off_shore,off_shore,255.47,shallow,True,0,2023,5,1


Day doesn't matter as earthquakes are random events. Hour is a relevant time scale for the human perception and can be used to measure the effects over its duration. Month and year help capture seasonal and annual cycle trends.

---

# Section 4 - Loading

Let's save what we've done into a new .csv file, ready to move onto to EDA in the next notebook. 

In [14]:
# Datafrom to csv
df_raw.to_csv("Dataset/Cleaned/earthquake_1995-2023_cleaned.csv", index=False)

---